lib/file_transfer_agent/encrypt_util.js

```diff
@@ -125,206 +124,278 @@ function EncryptUtil(encrypt, filestream, temp) {
     return newMatDesc;
   }
 
+  function createEncryptionMetadata(encryptionMaterial, keySize, encryptedKey, dataIv, keyIv = null, dataAad = null, keyAad = null) {
+    const matDesc = MaterialDescriptor(
+      encryptionMaterial.smkId,
+      encryptionMaterial.queryId,
+      keySize * 8
+    );
+
+    if (keyIv && dataAad && keyAad) {
+      keyIv = keyIv.toString(BASE64);
```

Comment:
should we rewrite our incoming parameters? isn't that antipattern?

Reply:
Yes, it was a bit smelly - changed it to ternary operator and moved to constructor args